assets/js/index.js

```diff
@@ -0,0 +1,262 @@
+const dashboardContainer = $("#dashboard-container");
+const cityList = $("#city-list");
+
+const API_KEY = "82c607cdbf31b750f2970e7092997d99";
+
+const getFormattedDate = (unixTimeStamp, format = "ddd DD/MM/YYYY") => {
+  return moment.unix(unixTimeStamp).format(format);
+};
+
+const getCurrentData = (name, forecastData) => {
+  return {
+    name: name,
+    temperature: forecastData.current.temp,
+    feels_like: forecastData.current.feels_like,
+    wind: forecastData.current.wind_speed,
+    humidity: forecastData.current.humidity,
+    uvi: forecastData.current.uvi,
+    date: getFormattedDate(forecastData.current.dt),
+    iconCode: forecastData.current.weather[0].icon,
+  };
+};
+
+const getForecastData = (forecastData) => {
+  const callBack = (each) => {
+    return {
+      date: getFormattedDate(each.dt),
+      max_temperature: each.temp.max,
+      min_temperature: each.temp.min,
+      wind: each.wind_speed,
+      humidity: each.humidity,
+      iconCode: each.weather[0].icon,
+    };
+  };
+
+  return forecastData.daily.slice(1, 8).map(callBack);
+};
+
+//get Data from API
+const getWeatherData = async (cityName) => {
+  const currentDataUrl = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${API_KEY}`;
+
+  const currentDataResponse = await fetch(currentDataUrl);
+
+  if (currentDataResponse.status === 200) {
+    const currentData = await currentDataResponse.json();
+
+    setCitiesInLs(cityName);
+
+    const lat = currentData.coord.lat;
+    const lon = currentData.coord.lon;
+    const name = currentData.name;
+
+    const forecastDataUrl = `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&appid=${API_KEY}&units=imperial`;
+
+    const forecastDataResponse = await fetch(forecastDataUrl);
+    const forecastData = await forecastDataResponse.json();
+
+    const current = getCurrentData(name, forecastData);
+
+    const forecast = getForecastData(forecastData);
+
+    return {
+      current: current,
+      forecast: forecast,
+    };
+  }
+};
+
+const getCitiesFromLs = () => {
+  //get cities from LS
```

Comment:
To improve this code even further, we can just directly return this line of code, rather than setting a variable just to return it:

`return JSON.parse(localStorage.getItem("recentCities")) ?? [];`

Reply:
Nice!!!